src/main/java/com/netflix/imflibrary/st2067_2/IMFCoreConstraintsChecker.java

```diff
@@ -53,6 +57,44 @@ public static List checkVirtualTracks(IMFCompositionPlaylistType compositionPlay
             else if(virtualTrack.getSequenceTypeEnum().equals(Composition.SequenceTypeEnum.MainAudioSequence)){
                 foundMainAudioEssence = true;
             }
+
+            if((virtualTrack.getSequenceTypeEnum().equals(Composition.SequenceTypeEnum.MainImageSequence)
+                    || virtualTrack.getSequenceTypeEnum().equals(Composition.SequenceTypeEnum.MainAudioSequence))
+                    && compositionPlaylistType.getEssenceDescriptorList() != null
+                    && compositionPlaylistType.getEssenceDescriptorList().size() > 0)
+            {
+                //Construct a DOMNodeObjectModel object for every EssenceDescriptor that is a part of the EssenceDescriptorList
+                Map<UUID, List<DOMNodeObjectModel>> domNodeObjectModelMap = new HashMap<>();
+                List<IMFEssenceDescriptorBaseType> essenceDescriptors = compositionPlaylistType.getEssenceDescriptorList();
+                for(IMFEssenceDescriptorBaseType essenceDescriptorBaseType : essenceDescriptors){
+                    List<DOMNodeObjectModel> domNodeObjectModels = new ArrayList<>();
+                    for (Object object : essenceDescriptorBaseType.getAny()) {
+                        domNodeObjectModels.add(new DOMNodeObjectModel((Node) object));
+                    }
+                    domNodeObjectModelMap.put(essenceDescriptorBaseType.getId(), domNodeObjectModels);
```

Comment:
getEssenceDescriptorListMap() is implemented in Composition guess I could pass in the Composition object to the IMFCoreConstraintsChecker() I just took the approach of not changing the interface to checkVirtualTracks() and construct the list internally.